PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionContentPanel.kt

```diff
@@ -228,11 +236,29 @@ open class SparkSubmissionContentPanel : JPanel() {
         toolTipText = refFilesPrompt.toolTipText
     }
 
-    val storageWithUploadPathPanel: SparkSubmissionJobUploadStorageWithUploadPathPanel =
+    var isStorageWithUploadPathPanelEnabled : Boolean = true
+
+    var storageWithUploadPathPanel: SparkSubmissionJobUploadStorageWithUploadPathPanel? =
         SparkSubmissionJobUploadStorageWithUploadPathPanel()
 
     val clusterSelectedSubject: BehaviorSubject<String> = BehaviorSubject.create<String>()
 
+    protected val sparkEventsLabel = JLabel("Spark Events:").apply {
```

Comment:
Can move these into Cosmos Serverless Spark?

Reply:
Agree with Wei, SparkEvents related codes can be put into sub-class `CosmosServerlessSparkSubmissionContentPanel`

---
In reply to: [234097840](https://github.com/Microsoft/azure-tools-for-java/pull/2353#discussion_r234097840) [](ancestors = 234097840)